### PR TITLE
[Metricbeat] Migrate ceph to reporter V2 with error handling

### PR DESCRIPTION
**Pull request by fearful-symmetry**
see #11374

This one is a tad large, as I took the time to add comments and change variable names in order to make the linters happy.

### CODE REVIEW

File: metricbeat/module/ceph/cluster_health/cluster_health.go [hunk 1]
```diff
@@ -42,11 +44,13 @@ func init() {
 	)
 }
 
+// MetricSet type defines all fields of the MetricSet
 type MetricSet struct {
 	mb.BaseMetricSet
 	*helper.HTTP
 }
 
+// New creates a new instance of the docker memory MetricSet.
```
Comment (jsoriano) [inline metricbeat/module/ceph/cluster_health/cluster_health.go]:
```suggestion
// New creates a new instance of the ceph cluster_health MetricSet.
```

File: metricbeat/module/ceph/cluster_status/cluster_status_test.go [hunk 2]
```diff
@@ -100,23 +100,23 @@ func TestFetchEventContents(t *testing.T) {
...
+
+	pgStateInfo = events[3].MetricSetFields["pg_state"].(common.MapStr)
+	assert.EqualValues(t, "active+remapped", pgStateInfo["state_name"])
+	assert.EqualValues(t, 55, pgStateInfo["count"])
+	assert.EqualValues(t, 813, pgStateInfo["version"])
+
+	pgStateInfo = events[4].MetricSetFields["pg_state"].(common.MapStr)
+	assert.EqualValues(t, "active+undersized+degraded+remapped", pgStateInfo["state_name"])
+	assert.EqualValues(t, 55, pgStateInfo["count"])
+	assert.EqualValues(t, 813, pgStateInfo["version"])
```
Comment (jsoriano) [inline metricbeat/module/ceph/cluster_status/cluster_status_test.go]: :+1:

File: metricbeat/module/ceph/cluster_health/data.go [hunk 3]
```diff
@@ -25,17 +25,20 @@ import (
 	"github.com/elastic/beats/libbeat/common"
 )
 
+//Timecheck contains part of the response from a HealthRequest
```
Comment (jsoriano) [inline metricbeat/module/ceph/cluster_health/data.go]: Spaces before comments please :slightly_smiling_face: 
```suggestion
// Timecheck contains part of the response from a HealthRequest
```

File: metricbeat/module/ceph/cluster_status/cluster_status.go [hunk 4]
```diff
@@ -42,11 +44,13 @@ func init() {
 	)
 }
 
+// MetricSet type defines all fields of the MetricSet
 type MetricSet struct {
 	mb.BaseMetricSet
 	*helper.HTTP
 }
 
+// New creates a new instance of the docker memory MetricSet.
```
Comment (jsoriano) [inline metricbeat/module/ceph/cluster_status/cluster_status.go]:
```suggestion
// New creates a new instance of the ceph cluster_status MetricSet.
```

File: metricbeat/module/ceph/osd_df/osd_df.go [hunk 5]
```diff
@@ -42,11 +44,13 @@ func init() {
 	)
 }
 
+// MetricSet type defines all fields of the MetricSet
 type MetricSet struct {
 	mb.BaseMetricSet
 	*helper.HTTP
 }
 
+// New creates a new instance of the docker memory MetricSet.
```
Comment (jsoriano) [inline metricbeat/module/ceph/osd_df/osd_df.go]: Docker also here

File: metricbeat/module/ceph/osd_tree/osd_tree.go [hunk 6]
```diff
@@ -42,11 +44,13 @@ func init() {
 	)
 }
 
+// MetricSet type defines all fields of the MetricSet
 type MetricSet struct {
 	mb.BaseMetricSet
 	*helper.HTTP
 }
 
+// New creates a new instance of the docker memory MetricSet.
```
Comment (jsoriano) [inline metricbeat/module/ceph/osd_tree/osd_tree.go]: And review this comment too.

File: metricbeat/module/ceph/pool_disk/pool_disk.go [hunk 7]
```diff
@@ -42,11 +44,13 @@ func init() {
 	)
 }
 
+// MetricSet type defines all fields of the MetricSet
 type MetricSet struct {
 	mb.BaseMetricSet
 	*helper.HTTP
 }
 
+// New creates a new instance of the docker memory MetricSet.
```
Comment (jsoriano) [inline metricbeat/module/ceph/pool_disk/pool_disk.go]: And this one.

**Comment by fearful-symmetry** [comment]
jenkins, test this

File: metricbeat/module/ceph/cluster_status/cluster_status_test.go (hunk 2 above)
Comment (kaiyan-sheng) [reply metricbeat/module/ceph/cluster_status/cluster_status_test.go]: 👍

File: metricbeat/module/ceph/osd_tree/osd_tree.go [hunk 8]
```diff
@@ -63,24 +67,20 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
...
 	events, err := eventsMapping(content)
 	if err != nil {
-		m.Logger().Error(err)
-		reporter.Error(err)
-		return
+		return errors.Wrap(err, "error in mapping")
 	}
 
 	for _, event := range events {
 		reporter.Event(mb.Event{MetricSetFields: event})
```
Comment (kaiyan-sheng) [inline metricbeat/module/ceph/osd_tree/osd_tree.go]: Add `if !reported { return nil }` check to be consistent with https://github.com/elastic/beats/pull/11979/files#diff-dc97d499b303003462f444c7164c2780R78?

**Pull request merged**